### PR TITLE
Dell S6100 component API fix

### PR DESCRIPTION
**Pull request by aravindmani-1**
#### Why I did it
sonic-mgmt test failure is seen for update_firmware component API
#### How I did it
Edited API 2.0 to fix this issue.
#### How to verify it
Run sonic-mgmt test after the fix and verify it passes.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205
- [x] 202211
- [x] 202305

### CODE REVIEW

**Comment by aravindmani-1** [comment]
we will support individual component upgrade once the request comes officially.

File: platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/component.py [hunk 1]
```diff
@@ -321,3 +321,16 @@ def install_firmware(self, image_path):
             A boolean, True if install was successful, False if not
         """
         return False
+
+    def update_firmware(self,image_path):
```
[the remark this answers is not in the document]
Comment (aravindmani-1) [reply platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/component.py:325]: We will support individual component upgrade once the support comes officially. 
https://github.com/sonic-net/sonic-buildimage/issues/16587 has been raised to track this issue.

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@StormLiangMS please cherry pick to 202305

**Comment by StormLiangMS** [comment]
ADO: 24709430

**Comment by prgeor** [comment]
@aravindmani-1 can you check if we need to cherry pick this to 202012? I see cherry pick conflict?

**Comment by aravindmani-1** [comment]
> @aravindmani-1 can you check if we need to cherry pick this to 202012? I see cherry pick conflict?

The fix is already available in 202012 branch.https://github.com/sonic-net/sonic-buildimage/blob/2042f44d973822fce3ee7d3deaee6b050ef295c2/platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/component.py#L325

**Comment by liuh-80** [comment]
Add 202012 label because already included in 202012

**Pull request merged**